pycbc/inference/io/dynesty.py

```diff
@@ -132,7 +132,12 @@ def read_raw_samples(self, fields, raw_samples=False, seed=0):
                     i += 1
                 else:
                     j += 1
-            rng = numpy.random.default_rng(seed)
+            try:
+                rng = numpy.random.default_rng(seed)
+            except AttributeError:
```

Comment:
This doesn't relate to python2.7 and it may lead us to trouble if you use a different random source with a try except. Please remove the try except and the comment.

Reply:
It relates to python 2.7 in the sense that you cannot use numpy version >= 1.17 if you are using python 2.7. When we drop 2.7 support, we can just bump the minimum numpy version to 1.17, and this issue goes away. I don't understand your second point about using a different random source. All this is being used for is to generate a one time random number generator for doing the shuffle, so as not to mess with the global generator that may be used elsewhere in whatever executable this is being used in. Are you worried that an AttributeError may be raised for another reason then just an old numpy version? If so, then I can add a check on the numpy version before trying `RandomState`; e.g.,
```
try:
    rng = numpy.random.default_rng(seed)
except AttributeError as e:
   if numpy.__version__ < 1.17:  # this is pseudo code
        rng = numpy.random.RandomState(seed)
   else:
        raise e
```
Would you prefer that?

Either way, I'd rather not just use `RandomState` since it states in the [numpy documentation](https://numpy.org/doc/stable/reference/random/legacy.html#numpy.random.RandomState) that it is only being kept for legacy purposes.

Reply:
@cdcapano The main issue is that you have two different paths that give different results. If you can use the Generator interface *which matches the output of Randomstate* then I would not object, but I think your default setup that's not guaranteed. I don't want someone to get a different result depending on if they use python2 / 3 when that is not needed.

Reply:
Ok, I understand your concern now. Yeah, with the current patch you'll end up with slightly different results if you use different versions of numpy. The only way to get exactly the same results is to use `RandomState` (you'll get different results from `RandomState` even if you use the MT19937 generator, even though it's nominally the same algorithm). But the current [numpy docs](https://numpy.org/doc/stable/reference/random/index.html) state that you should use `default_rng` for creating a generator. If they drop support for `RandomState` sometime in the future, then we'll need to change what's used... and at that point, you'll end up with different results if you use different versions of the code. Why not just do that now? At least the python 2.7/3.X thing is a clear deliminator. Besides, isn't expecting that you get exactly the same realization of random numbers regardless of what version of code you use a bit strict?